collector/osv/src/server.rs

```diff
@@ -130,22 +135,18 @@ pub async fn gather(
             }
         }
     }
-    let purls: Vec<_> = response
+    let purls: HashMap<_, _> = response
         .results
         .iter()
-        .flat_map(|e| {
-            if let Package::Purl { purl } = &e.package {
-                Some(purl)
-            } else {
-                None
+        .flat_map(|e| match (&e.package, &e.vulns) {
+            (Package::Purl { purl }, Some(v)) if !v.is_empty() => {
```

Comment:
v could be Some([]) or None? Can we normalize that higher so we don't have empty arrays inside a Some? Or am I mis-reading?

Reply:
i think the `if !v.is_empty()` qualifier accounts for that, but i need to have a test in there to confirm.